firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/Personalization.java

```diff
@@ -16,22 +16,41 @@
 
 import android.os.Bundle;
 import androidx.annotation.NonNull;
+import com.google.common.base.CaseFormat;
+import com.google.common.base.Converter;
 import com.google.firebase.analytics.connector.AnalyticsConnector;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 import org.json.JSONObject;
 
 public class Personalization {
   public static final String ANALYTICS_ORIGIN_PERSONALIZATION = "fp";
-  public static final String ANALYTICS_PULL_EVENT = "_fpc";
-  public static final String ARM_KEY = "_fpid";
-  public static final String ARM_VALUE = "_fpct";
-  static final String PERSONALIZATION_ID = "personalizationId";
+
+  public static final String ANALYTICS_PULL_EVENT = "personalization_choice";
+  public static final String ARM_KEY = "arm_key";
+  public static final String ARM_VALUE = "arm_value";
+  public static final String PERSONALIZATION_ID = "personalization_id";
+  public static final String ARM_INDEX = "arm_index";
+  public static final String GROUP = "group";
+
+  public static final String ANALYTICS_PULL_EVENT_INTERNAL = "_fpc";
+  public static final String CHOICE_ID = "choiceId";
+  public static final String CHOICE_ID_KEY = "_fpid";
+
+  private static final Converter<String, String> CONVERTER =
```

Comment:
Do we need to use a `Converter` for this? Seems like we're just converting constants and it adds the guava dependency.

Reply:
Removed the `Converter`. Is Guava a heavy dependency?